### ISSUE TITLE
docs: tls secrets must be namespaced

### ISSUE THREAD

**Issue by mythi** [jcmoraisjr/haproxy-ingress#253]
Since the deployment .yamls add

```
--default-ssl-certificate=$(POD_NAMESPACE)/tls-secret
```

the docs should guide to create tls secrets in POD_NAMESPACE too. Essentially:

```
$ kubectl create secret tls tls-secret --key tls.key --cert tls.crt -n ingress-controller
```

Ditto for foobar-ssl?

**Comment by jcmoraisjr** [comment]
Indeed!

**Comment by rikatz** [comment]
Making a PR here to improve some docs issues

**Comment by koenw** [comment]
Should tls secrets also be namespaced in the `Ingress` resource? I'm asking because I've got a case where haproxy always falls back to the default certificate (However, cert-manager complains if I prefix the `secretName` in the `Ingress` resource with a namespace :disappointed:  ).

**Comment by bugfest** [comment]
Hi @koenw - just faced the same issue. In my case It was due to an incorrect certificate format. After some debugging I found the controller was complaining about it and quietly falling back to the default one:

```
I0923 12:15:22.918810       6 ssl.go:58] Creating temp file /ingress-controller/ssl/default-my.example-tls.pem515874826 for Keypair: default-my.example-tls.pem
I0923 12:15:22.918920       6 backend_ssl.go:50] syncing a non ca/crt secret default/my.example-tls
W0923 12:15:22.919289       6 controller.go:1289] ssl certificate "default/my.example-tls" does not exist in local store. Using the default cert
```
Good luck!

**Comment by jcmoraisjr** [comment]
Addressed since #277, closing.
